trace-utils/src/send_data/retry_strategy.rs

```diff
@@ -0,0 +1,228 @@
+// Copyright 2023-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use std::time::Duration;
+use tokio::time::sleep;
+
+/// Enum representing the type of backoff to use for the delay between retries.
+/// ```
+#[derive(Debug, Clone)]
+pub enum RetryBackoffType {
+    /// Increases the delay by a fixed increment each attempt.
+    Linear,
+    /// The delay is constant for each attempt.
+    Constant,
+    /// The delay is doubled for each attempt.
+    Exponential,
+}
+
+// TODO: APMSP-1076 - RetryStrategy should be moved to a separate file when send_data is refactored.
+/// Struct representing the retry strategy for sending data.
+///
+/// This struct contains the parameters that define how retries should be handled when sending data.
+/// It includes the maximum number of retries, the delay between retries, the type of backoff to
+/// use, and an optional jitter to add randomness to the delay.
+///
+/// # Examples
+///
+/// ```rust
+/// use datadog_trace_utils::send_data::{RetryBackoffType, RetryStrategy};
+/// use std::time::Duration;
+///
+/// let retry_strategy = RetryStrategy {
+///     max_retries: 5,
+///     delay_ms: Duration::from_millis(100),
+///     backoff_type: RetryBackoffType::Exponential,
+///     jitter: Some(Duration::from_millis(50)),
+/// };
+/// ```
+#[derive(Debug, Clone)]
+pub struct RetryStrategy {
+    /// The maximum number of retries to attempt.
+    pub max_retries: u32,
+    /// The minimum delay between retries.
+    pub delay_ms: Duration,
+    /// The type of backoff to use for the delay between retries.
+    pub backoff_type: RetryBackoffType,
+    /// An optional jitter to add randomness to the delay.
+    pub jitter: Option<Duration>,
```

Comment:
Should we limit the visibility of the fields to the crate?

Reply:
I'd have to implement  `new()` for RetryStrategy...which is probably a good idea.